### PR TITLE
BaSP-TG-23: Test Projects: put, delete

### PR DESCRIPTION
**Pull request by silveriac**
Only tests for status 200, 204, 400 (only tests one validation) and 404.

### CODE REVIEW

File: src/test/projects.test.js [hunk 1]
```diff
@@ -1 +1,53 @@
...
+import app from '../app';
+import Project from '../models/Projects';
+import projectSeed from '../seed/projects';
+
+beforeAll(async () => {
+  await Project.collection.insertMany(projectSeed);
+});
+
+const projectId = '63531aaa2b654a3fb77054dd';
+const seed = {
```
Comment (MaicolFreire001) [inline src/test/projects.test.js:11]: We should call this const "mockedProject"

File: src/test/projects.test.js [hunk 2]
```diff
@@ -1 +1,53 @@
...
+
+describe('DELETE /project/:id', () => {
+  test('everything correct: returns status code 204', async () => {
+    const res = await request(app).delete(`/projects/${projectId}`);
+    expect(res.status).toBe(200);
+  });
+  test('wrong id: returns status code 404', async () => {
+    const res = await request(app).delete('/projects/69');
+    expect(res.status).toBe(404);
+  });
```
Comment (MaicolFreire001) [inline src/test/projects.test.js:52]: We could also test if the id is invalid

File: src/test/projects.test.js [hunk 3]
```diff
@@ -1 +1,53 @@
...
+  test('wrong id: returns status code 404', async () => {
+    const res = await request(app).put('/projects/69').send(seed);
+    expect(res.status).toBe(404);
+  });
+  test('validations: returns status code 400', async () => {
+    seed.name = '';
+    const res = await request(app).put(`/projects/${projectId}`).send(seed);
+    expect(res.status).toBe(400);
+    expect(res.body.message[0].message).toBe('Name required');
+  });
```
Comment (MaicolFreire001) [inline src/test/projects.test.js:41]: We could also test if the id is invalid

**Review by MaicolFreire001 — CHANGES_REQUESTED** [review]

File: src/test/projects.test.js (hunk 2 above)
Comment (silveriac) [reply src/test/projects.test.js:52]: Are you talking about the returned error message?

Comment (MaicolFreire001) [reply src/test/projects.test.js:52]: Yes but also a separate test for invalid id and wrong id

Comment (silveriac) [reply src/test/projects.test.js:52]: Agreed. I'll add it

File: src/test/projects.test.js [hunk 4]
```diff
@@ -1 +1,63 @@
...
+  test('invalid id: return status code 400', async () => {
+    const res = await request(app).delete('/projects/42');
+    expect(res.status).toBe(400);
+    expect(res.body.message).toBe('Invalid id 42.');
+  });
+  test('id does not exist: return status code 404', async () => {
+    const res = await request(app).delete(`/projects/${mockedProject.teamMembers[0].employee}`);
+    expect(res.status).toBe(404);
+  });
```
Comment (MaicolFreire001) [inline src/test/projects.test.js:62]: We could add and expect for the message and/or error boolean

File: src/test/projects.test.js [hunk 5]
```diff
@@ -1 +1,63 @@
...
+  test('invalid id: return status code 400', async () => {
+    mockedProject.name = 'Eric';
+    const res = await request(app).put('/projects/69').send(mockedProject);
+    expect(res.status).toBe(400);
+  });
+  test('id does not exist: return status code 404', async () => {
+    const res = await request(app).put(`/projects/${mockedProject.teamMembers[0].employee}`).send(mockedProject);
+    expect(res.status).toBe(404);
+  });
```
Comment (MaicolFreire001) [inline src/test/projects.test.js:46]: We could add and expect for the message and/or error boolean

File: src/test/projects.test.js [hunk 6]
```diff
@@ -1 +1,63 @@
...
+  test('validations: return status code 400', async () => {
+    mockedProject.name = '';
+    const res = await request(app).put(`/projects/${projectId}`).send(mockedProject);
+    expect(res.status).toBe(400);
+    expect(res.body.message[0].message).toBe('Name required');
+  });
+  test('invalid id: return status code 400', async () => {
+    mockedProject.name = 'Eric';
+    const res = await request(app).put('/projects/69').send(mockedProject);
+    expect(res.status).toBe(400);
```
Comment (MaicolFreire001) [inline src/test/projects.test.js:41]: We could add and expect for the message and/or error boolean

File: src/test/projects.test.js [hunk 7]
```diff
@@ -1 +1,63 @@
...
+  test('everything correct: return status code 200', async () => {
+    const res = await request(app).put(`/projects/${projectId}`).send(mockedProject);
+    expect(res.status).toBe(200);
+  });
+  test('validations: return status code 400', async () => {
+    mockedProject.name = '';
+    const res = await request(app).put(`/projects/${projectId}`).send(mockedProject);
+    expect(res.status).toBe(400);
+    expect(res.body.message[0].message).toBe('Name required');
+  });
```
Comment (MaicolFreire001) [inline src/test/projects.test.js:37]: We could add and expect for the error boolean

File: src/test/projects.test.js [hunk 8]
```diff
@@ -1 +1,63 @@
...
+      rate: 100,
+    },
+  ],
+};
+
+describe('PUT /project/:id', () => {
+  test('everything correct: return status code 200', async () => {
+    const res = await request(app).put(`/projects/${projectId}`).send(mockedProject);
+    expect(res.status).toBe(200);
+  });
```
Comment (MaicolFreire001) [inline src/test/projects.test.js:31]: We could add and expect for the message and/or error boolean

**Review by MaicolFreire001 — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 9]
```diff
@@ -3,7 +3,7 @@ import app from './app';
 
 require('dotenv').config();
 
-const port = process.env.PORT || 3000;
+const port = process.env.PORT || 4000;
```
Comment (nicolasblasco) [inline src/index.js:6]: We should change the port value

File: src/test/projects.test.js [hunk 10]
```diff
@@ -1 +1,69 @@
...
+      role: 'DEV',
+      rate: 100,
+    },
+  ],
+};
+
+describe('PUT /project/:id', () => {
+  test('everything correct: return status code 200', async () => {
+    const res = await request(app).put(`/projects/${projectId}`).send(mockedProject);
+    expect(res.status).toBe(200);
```
Comment (nicolasblasco) [inline src/test/projects.test.js:30]: We should test the message and the error in the response

File: src/test/projects.test.js [hunk 11]
```diff
@@ -1 +1,69 @@
...
+    const res = await request(app).put(`/projects/${projectId}`).send(mockedProject);
+    expect(res.status).toBe(400);
+    expect(res.body.message[0].message).toBe('Name required');
+    expect(res.body.error).toBeTruthy();
+  });
+  test('invalid id: return status code 400', async () => {
+    mockedProject.name = 'Eric';
+    const res = await request(app).put('/projects/69').send(mockedProject);
+    expect(res.status).toBe(400);
+    expect(res.body.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/test/projects.test.js:43]: We should also test the message of the body in the response

File: src/test/projects.test.js [hunk 12]
```diff
@@ -1 +1,69 @@
...
+  });
+  test('id does not exist: return status code 404', async () => {
+    const res = await request(app).put(`/projects/${mockedProject.teamMembers[0].employee}`).send(mockedProject);
+    expect(res.status).toBe(404);
+    expect(res.body.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/test/projects.test.js:48]: We should also test the message of the body in the response

**Review by nicolasblasco — CHANGES_REQUESTED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Review by MaicolFreire001 — APPROVED** [review]

**Pull request merged**